recipes-bsp/imx-atf/imx-atf_2.6.bb

```diff
@@ -9,10 +9,10 @@ FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
 
 PV .= "+git${SRCPV}"
 
-SRCBRANCH = "lf_v2.4"
+SRCBRANCH = "lf_v2.6"
 SRC_URI = "git://source.codeaurora.org/external/imx/imx-atf.git;protocol=https;branch=${SRCBRANCH} \
            file://0001-Makefile-Suppress-array-bounds-error.patch"
-SRCREV = "05f788b9bbb13d002997e35008a4b945f7e2957b"
+SRCREV = "c6a19b1a351308cc73443283f6aa56b2eff791b8"
```

Comment:
@thochstein, please send the `0001-Makefile-Suppress-array-bounds-error.patch` for internal review. Ideally, those warnings ought to be fixed but merging this also helps.

Reply:
As the commit message says, the patch was only meant to be temporary. It looks like we never needed it. Please try just dropping it.

Reply:
I did and it fails build using new GCC, so yes, it is required.

Reply:
Oh, strange, I wonder why we do not have the problem.

Reply:
It fails if using master. Please check.

Reply:
The warnings are false-positives. The fix commit is shown here:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578#c39

As the commit says, the fix for GCC 12 is not a complete fix, and code that uses a constant less than 4kB as an address --which is unfortunately the case here -- will still trigger the warning. For GCC 13 it appears that the false-positives are completely eliminated by looking more specifically for a NULL value.

So I don't think there's anything to do but to use this workaround for now and wait for the fix in GCC 13.

Reply:
Would you mind sending a PR adding this information to the patch file?

Reply:
I reviewed the comment and it looks fine. @kraj have this issue been fixed in GCC13? Could this be backported?

Reply:
yes it is ok in gcc13 and also in clang14

Reply:
Could it be backported to kirkstone? or not?

Reply:
There's one thing still bothering me though. The bug mentions fixing GCC 11 as well. We are building imx-atf from our meta-imx layer with GCC 11 without any problem with array-bounds. Does it make sense?

Reply:
> Could it be backported to kirkstone? or not?

You'd think if they could have done the complete solution in GCC 12 they would have?

Reply:
its not an easy backport since it need infra which is only added in gcc13

Reply:
> 

its fixed/backported partially into gcc 11.3 and if you are using latest kirkstone then we are on 11.3

Reply:
So in the case of imx-atf, the false positives we see are not fixed by the partial fix since the constants are less than 4kB. So what's strange is we do not see the array-bounds error.

Reply:
> Would you mind sending a PR adding this information to the patch file?

See #1142.